packages/store-ui/src/molecules/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,73 @@
+import type {
+  ChangeEvent,
+  FormEvent,
+  InputHTMLAttributes,
+  ReactNode,
+} from 'react'
+import React, { forwardRef, useState } from 'react'
+
+import Button from '../../atoms/Button'
+import Icon from '../../atoms/Icon'
+import Input from '../../atoms/Input'
+
+const SearchIcon = () => (
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    width="1em"
+    height="1em"
+    viewBox="0 0 24 24"
+    fill="none"
+    stroke="currentColor"
+    strokeWidth="2"
+    strokeLinecap="round"
+    strokeLinejoin="round"
+  >
+    <circle cx="11" cy="11" r="8" />
+    <line x1="21" y1="21" x2="16.65" y2="16.65" />
+  </svg>
+)
+
+type InputProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'onSubmit'>
+
+export interface SearchInputProps extends InputProps {
+  onSubmit: (value: string) => void
+  icon?: ReactNode
+  testId?: string
+}
+
+const SearchInput = forwardRef<HTMLFormElement, SearchInputProps>(
+  function SearchInput(
+    { onSubmit, icon, testId = 'store-search-input', ...props },
+    ref
+  ) {
+    const [search, setSearch] = useState<string>('')
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+
+      if (search !== '') {
+        onSubmit(search)
+      }
+    }
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+      setSearch(event.target.value)
+    }
+
+    return (
+      <form
+        ref={ref}
+        data-store-search-input
+        data-testid={testId}
+        onSubmit={handleSubmit}
+      >
+        <Input name="search" onChange={handleChange} {...props} />
+        <Button type="submit">
+          <Icon component={icon ?? <SearchIcon />} />
```

Comment:
why don't we use the `children` in here?

Reply:
so, this is one of the things you can do with this component. We can call the event `onFocus` from the input to open the field of suggestions/results. But this should not be done in this component, because as his name already says: it is just the Input for search.

Reply:
this is an example for it: https://codesandbox.io/s/store-ui-search-input-styling-nijuh?file=/src/App.tsx

Reply:
nice! I haven't seen these props spread, hehe

Reply:
But again. Why doesn't we use something like?
```suggestion
          <Icon><SearchIcon /></Icon>
```

Reply:
Our Icon component does not support children's prop. And he was inspired by Ant Design Icon, but if you think that this way is better, we can make the change. But it's not in this pr.